lib/wibox/layout/fixed.lua

```diff
@@ -23,10 +23,15 @@ local fixed = {}
 function fixed:layout(context, width, height)
     local result = {}
     local pos,spacing = 0, self._private.spacing
+    local spacing_widget = self._private.spacing_widget
+    local is_y = self._private.dir == "y"
+    local is_x = not is_y
+    local abspace = math.abs(spacing)
```

Comment:
Wait, I can have negative spacing between widgets and can show a spacing widget in this non-existent space. Why??

Reply:
See the image above. The powerline example uses it. A while ago we had a discussion and 3 or 4 prototypes how to best create powerlines. However the negative spacing trick is actually from the screenshot page. I found how he/she was quite right, it does work rather well expect for the mouse area issue. The last PR of this series have an example in the doc for the "true" powerline taglist using negative spacing.

![screenshot_20171124_035338](https://user-images.githubusercontent.com/340384/33202583-0f283446-d0cb-11e7-9559-e2e30d70afb3.png)